Verify selectionsort

dario23 opened the pull request.

dario23 committed:
Add successfully verifying selection sort testcase

dario23 committed:
Trigger on lookup_pure as well in lookup_pure equality

Aurel300 commented on prusti-tests/tests/verify/pass/arrays/selection_sort.rs:
```diff
@@ -0,0 +1,56 @@
+fn main() {}
+
+
+// TODO: replace 10 with a.len()
```
Is this something that will be part of this PR?

dario23 replied:
that's a relict from me writing the test before #538 was merged, and now that i checked, it'll also need more slice snapshot support for using it in the invariants (unsize cast, `.len()` call, .. -> #564). For the non-pure part (i.e. the loop conditions) we can use `.len()` already, so i'll change that.

dario23 committed:
Use a.len() for non-specification code

For specification code, we need more slice support in the pure function
encoder.
